From 9759af04a0fad70219310575980eef840561163b Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 26 Dec 2023 23:20:22 -0600
Subject: [PATCH] Hack it to work in Steam properly

If we do this then Steam will properly inject both libs into the containerized environment

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 data/gamemoderun | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/data/gamemoderun b/data/gamemoderun
index 573b3e4..f4ba8c7 100755
--- a/data/gamemoderun
+++ b/data/gamemoderun
@@ -2,8 +2,9 @@
 # Helper script to launch games with gamemode
 
 GAMEMODEAUTO_NAME="libgamemodeauto.so.0"
+GAMEMODE_NAME="libgamemode.so.0"
 
 # ld will find the right path to load the library, including for 32-bit apps.
-LD_PRELOAD="${GAMEMODEAUTO_NAME}${LD_PRELOAD:+:$LD_PRELOAD}"
+LD_PRELOAD="${GAMEMODE_NAME}:${GAMEMODEAUTO_NAME}${LD_PRELOAD:+:$LD_PRELOAD}"
 
 exec env LD_PRELOAD="${LD_PRELOAD}" $GAMEMODERUNEXEC "$@"
-- 
2.43.0

